pkgs/tools/package-management/nix/default.nix

```diff
@@ -47,59 +47,31 @@ in lib.makeExtensible (self: {
     };
   }).override { boehmgc = boehmgc-nix_2_3; };
 
-  nix_2_4 = common {
-    version = "2.4";
-    sha256 = "sha256-op48CCDgLHK0qV1Batz4Ln5FqBiRjlE6qHTiZgt3b6k=";
-    # https://github.com/NixOS/nix/pull/5537
-    patches = [ ./patches/install-nlohmann_json-headers.patch ];
-  };
+  nix_2_4 = throw "nixVersions.nix_2_4 has been removed";
```

Comment:
Shouldn't these be in aliases.nix?

Reply:
There would be direct references to `nixVersions.*` out there, leave these throws here makes me feel more comfortable.